itest/lnd_funding_test.go

```diff
@@ -648,6 +656,139 @@ func testExternalFundingChanPoint(ht *lntest.HarnessTest) {
+	ht.FundCoins(btcutil.SatoshiPerBitcoin, carol)
+
+	// Before we start the test, we'll ensure both sides are connected to
+	// the funding flow can properly be executed.
+	ht.EnsureConnected(carol, dave)
+
+	// At this point, we're ready to simulate our external channel funding
+	// flow. To start with, we'll create a pending channel with a shim for
+	// a transaction that will never be published.
+	const chanSize = funding.MaxBtcFundingAmount
+	fundingShim1, chanPoint1 := deriveFundingShim(
+		ht, carol, dave, chanSize, 0, false, commitmentType,
+	)
+	ht.OpenChannelAssertPending(carol, dave, lntest.OpenChannelParams{
+		Amt:            chanSize,
+		FundingShim:    fundingShim1,
+		CommitmentType: commitmentType,
+		Private:        true,
+	})
+	ht.AssertNodesNumPendingOpenChannels(carol, dave, 1)
+
+	// That channel is now pending forever and normally would saturate the
+	// max pending channel limit for both nodes. But because the channel is
+	// externally funded, we should still be able to open another one. Let's
+	// do exactly that now. For this one we publish the transaction so we
+	// can mine it later.
+	fundingShim2, chanPoint2 := deriveFundingShim(
+		ht, carol, dave, chanSize, 0, true, commitmentType,
+	)
+
+	// At this point, we'll now carry out the normal basic channel funding
+	// test as everything should now proceed as normal (a regular channel
+	// funding flow).
+	carolChan, daveChan, _ := basicChannelFundingTest(
+		ht, carol, dave, fundingShim2, true, &commitmentType,
+	)
+
+	// The itest harness doesn't mine blocks for private channels, so we
+	// want to make sure the channel with the published and mined
+	// transaction leaves the pending state.
+	ht.MineBlocks(6)
+
+	rpcChanPointToStr := func(cp *lnrpc.ChannelPoint) string {
+		txid, err := chainhash.NewHash(cp.GetFundingTxidBytes())
+		require.NoError(ht, err)
+		return fmt.Sprintf("%v:%d", txid.String(), cp.OutputIndex)
+	}
+
+	pendingCarol := carol.RPC.PendingChannels().PendingOpenChannels
+	require.Len(ht, pendingCarol, 1)
+	require.Equal(
+		ht, rpcChanPointToStr(chanPoint1),
+		pendingCarol[0].Channel.ChannelPoint,
+	)
+	openCarol := carol.RPC.ListChannels(&lnrpc.ListChannelsRequest{
+		ActiveOnly:  true,
+		PrivateOnly: true,
+	})
+	require.Len(ht, openCarol.Channels, 1)
+	require.Equal(
+		ht, rpcChanPointToStr(chanPoint2),
+		openCarol.Channels[0].ChannelPoint,
+	)
+
+	pendingDave := dave.RPC.PendingChannels().PendingOpenChannels
+	require.Len(ht, pendingDave, 1)
+	require.Equal(
+		ht, rpcChanPointToStr(chanPoint1),
+		pendingDave[0].Channel.ChannelPoint,
+	)
+
+	openDave := dave.RPC.ListChannels(&lnrpc.ListChannelsRequest{
+		ActiveOnly:  true,
+		PrivateOnly: true,
+	})
+	require.Len(ht, openDave.Channels, 1)
+	require.Equal(
+		ht, rpcChanPointToStr(chanPoint2),
+		openDave.Channels[0].ChannelPoint,
+	)
+
+	// Both channels should be marked as frozen with the proper thaw height.
+	require.Equal(ht, 0, carolChan.ThawHeight, "thaw height")
```

Comment:
looks like we need to use `require.EqualValues` here

Reply:
Oops, thanks.